crates/oxc_minifier/src/peephole/minimize_statements.rs

```diff
@@ -1205,13 +1232,28 @@ impl<'a> PeepholeOptimizations {
         ctx: &Ctx<'a, '_>,
         non_scoped_literal_only: bool,
     ) -> usize {
+        // For large numbers of declarators, lazily collect all identifier names referenced
+        // in the target expression. This allows us to quickly skip declarators whose names
+        // aren't even present, avoiding expensive tree traversals.
+        // For small numbers of declarators, the overhead of creating the HashSet isn't worth it.
+        const IDENTIFIER_CACHE_THRESHOLD: usize = 4;
```

Comment:
this is a bit of a hack, but if we're minifying a small file e.g. the files in our benchmarks, creating the ident cache introduces a slowdown and has a negative 5+% perf hit. But if we've got a ton of idents like the one in the linked issue, it's worth it